### PR TITLE
DUPLO-20994 TF provider: Azure: Storage Account: Add support for  Blobs, Queues and Tables.

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Implementation of storage class resources
## Summary of changes
Implementation of 'duplocloud_azure_storageclass_blob', 'duplocloud_azure_storageclass_queue', 'duplocloud_azure_storageclass_table' resources
This PR does the following:

- Implemented CURD operations of 'duplocloud_azure_storageclass_blob', 'duplocloud_azure_storageclass_queue', 'duplocloud_azure_storageclass_table' resources
- Added example
- Updated documentation

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-20994 TF provider: Azure: Storage Account: Add support for  Blobs, Queues and Tables.](https://app.clickup.com/t/8655600/DUPLO-20994)

File: duplocloud/resource_duplo_azure_storage_blob.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			Description: "The URL of the Blob.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+	}
+}
+
+func resourceAzureStorageBlob() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_azure_storageclass_blob` manages an Azure storage share file in Duplo.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_blob.go:46]: Change the description.

File: duplocloud/resource_duplo_azure_storage_queue.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+func duploAzureStorageQueueSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the azure storage account share file will be created in.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+		"storage_account_name": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_queue.go:24]: Add `ForceNew` for storage_account_name.

File: duplocloud/resource_duplo_azure_storage_table.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+func duploAzureStorageTableSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the azure storage account share file will be created in.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+		"storage_account_name": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_table.go:24]: Add `ForceNew` for storage_account_name.

File: duplocloud/resource_duplo_azure_storage_blob.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func duploAzureStorageBlobSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the azure storage account share file will be created in.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_blob.go:19]: Please change wherever share file is used with actual resource.

File: duplocloud/resource_duplo_azure_storage_blob.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	if duplo == nil {
+		d.SetId("") // object missing
+		return nil
+	}
+	if clientErr != nil {
+		if clientErr.Status() == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Unable to retrieve tenant %s azure storage share file %s : %s", tenantID, name, clientErr)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_blob.go:82]: Change the log message for share file.

File: duplocloud/resource_duplo_azure_storage_blob.go [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+	tenantID, storageAccountName, name, err := parseAzureStorageResourceIdParts(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[TRACE] resourceAzureStorageBlobDelete(%s, %s, %s): start", tenantID, storageAccountName, name)
+
+	c := m.(*duplosdk.Client)
+	err = c.AzureStorageAccountBlobDelete(tenantID, storageAccountName, name)
+	if err != nil {
+		return diag.Errorf("Error creating tenant %s azure storage account share file '%s': %s", tenantID, name, err)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_blob.go:138]: Change the log message for share file.

File: duplocloud/resource_duplo_azure_storage_queue.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func duploAzureStorageQueueSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the azure storage account share file will be created in.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_queue.go:18]: Change the description for share file.

File: duplocloud/resource_duplo_azure_storage_queue.go [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+			Description: "The URL of the Queue.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+	}
+}
+
+func resourceAzureStorageQueue() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_azure_storageclass_queue` manages an Azure storage share file in Duplo.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_queue.go:45]: Change the description for share file.

File: duplocloud/resource_duplo_azure_storage_queue.go [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+	if duplo == nil {
+		d.SetId("") // object missing
+		return nil
+	}
+	if clientErr != nil {
+		if clientErr.Status() == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Unable to retrieve tenant %s azure storage share file %s : %s", tenantID, name, clientErr)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_queue.go:81]: Change the log message for share file.

File: duplocloud/resource_duplo_azure_storage_queue.go [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+	tenantID, storageAccountName, name, err := parseAzureStorageResourceIdParts(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[TRACE] resourceAzureStorageQueueDelete(%s, %s, %s): start", tenantID, storageAccountName, name)
+
+	c := m.(*duplosdk.Client)
+	err = c.AzureStorageAccountQueueDelete(tenantID, storageAccountName, name)
+	if err != nil {
+		return diag.Errorf("Error creating tenant %s azure storage account share file '%s': %s", tenantID, name, err)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_queue.go:137]: Change the log message for share file.

File: duplocloud/resource_duplo_azure_storage_table.go [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func duploAzureStorageTableSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"tenant_id": {
+			Description:  "The GUID of the tenant that the azure storage account share file will be created in.",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_table.go:18]: Change the description for share file.

File: duplocloud/resource_duplo_azure_storage_table.go [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+	if duplo == nil {
+		d.SetId("") // object missing
+		return nil
+	}
+	if clientErr != nil {
+		if clientErr.Status() == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Unable to retrieve tenant %s azure storage share file %s : %s", tenantID, name, clientErr)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_table.go:81]: Change the log message for share file.

File: duplocloud/resource_duplo_azure_storage_table.go [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+	tenantID, storageAccountName, name, err := parseAzureStorageResourceIdParts(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[TRACE] resourceAzureStorageTableDelete(%s, %s, %s): start", tenantID, storageAccountName, name)
+
+	c := m.(*duplosdk.Client)
+	err = c.AzureStorageAccountTableDelete(tenantID, storageAccountName, name)
+	if err != nil {
+		return diag.Errorf("Error creating tenant %s azure storage account share file '%s': %s", tenantID, name, err)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_table.go:137]: Change the log message for share file.

File: examples/resources/duplocloud_azure_storage_blob/import.sh [hunk 14]
```diff
@@ -0,0 +1,5 @@
+# Example: Importing an existing Azure storage account share file
```
Comment (tahir-duplo) [inline examples/resources/duplocloud_azure_storage_blob/import.sh:1]: Change the description for share file.

File: examples/resources/duplocloud_azure_storage_queue/import.sh [hunk 15]
```diff
@@ -0,0 +1,5 @@
+# Example: Importing an existing Azure storage account share file
```
Comment (tahir-duplo) [inline examples/resources/duplocloud_azure_storage_queue/import.sh:1]: Change the description for share file.

File: duplocloud/resource_duplo_azure_storage_blob.go [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+	name := d.Get("name").(string)
+	log.Printf("[TRACE] resourceAzureStorageBlobCreate(%s, %s, %s): start", tenantID, storageAccountName, name)
+	c := m.(*duplosdk.Client)
+
+	err = c.AzureStorageAccountBlobCreate(tenantID, storageAccountName, name)
+	if err != nil {
+		return diag.Errorf("Error creating tenant %s azure storage account blob '%s': %s", tenantID, name, err)
+	}
+
+	id := fmt.Sprintf("%s/%s/%s/%s", tenantID, storageAccountName, "queue", name)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_azure_storage_blob.go:108]: We should use `blob` instead of `queue` here

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**